packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmDictionaryTests.kt

```diff
@@ -0,0 +1,619 @@
+/*
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.test.shared
+
+import io.realm.kotlin.MutableRealm
+import io.realm.kotlin.Realm
+import io.realm.kotlin.RealmConfiguration
+import io.realm.kotlin.entities.dictionary.RealmDictionaryContainer
+import io.realm.kotlin.ext.query
+import io.realm.kotlin.ext.realmDictionaryOf
+import io.realm.kotlin.ext.toRealmDictionary
+import io.realm.kotlin.query.find
+import io.realm.kotlin.test.platform.PlatformUtils
+import io.realm.kotlin.test.shared.util.ErrorCatcher
+import io.realm.kotlin.test.shared.util.GenericTypeSafetyManager
+import io.realm.kotlin.test.util.TypeDescriptor
+import io.realm.kotlin.types.ObjectId
+import io.realm.kotlin.types.RealmAny
+import io.realm.kotlin.types.RealmDictionary
+import io.realm.kotlin.types.RealmInstant
+import io.realm.kotlin.types.RealmObject
+import io.realm.kotlin.types.RealmUUID
+import org.mongodb.kbson.BsonObjectId
+import kotlin.reflect.KClassifier
+import kotlin.reflect.KMutableProperty1
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertContains
+import kotlin.test.assertContentEquals
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+class RealmDictionaryTests {
```

Comment:
Missing different assignment tests:

1. Assigning unmanaged dictionary to a manged object should implicitly copy
2. Assigning a managed dictionary to another object, so either copy or throw (what do we do for Lists here)

Reply:
We are also lacking tests for accessing methods on closed dictionaries, but not sure if we do this for other Collections right now either